### PR TITLE
Expose history replace in action editor

### PR DESCRIPTION
**Pull request by breakthestatic**
## Proposed change

The existing handler allows for the `navigate` action to use `history.replace`, but this is not exposed to the front-end. Exposing this allows functionality as noted in https://community.home-assistant.io/t/parametr-to-control-back-history-in-navigate-action/471028, i.e. replacing url hashes without adding to the stack.  I frequently use this to control different tv remotes, e.g. one main universal, with conditional rendering based on hash.  If I go back and forth 5 times, that adds five new pages to the history, but from a user's perspective, it's still the same page/view and going back should actually navigate to the url *before* all the hash changes.

I switched the rendering of the fields to use a form schema, as it seemed a bit cleaner than manually adding another prop specific to `navigate` actions. As part of that, I removed the pre-existing props & handlers related to `navigate`. From my testing and manually going through the logic, none of it seemed necessary anymore with the switch to form schema. However, if I missed any use (or edge) cases, please advise!

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml
type: button
tap_action:
  action: navigate
  navigation_path: '#view-a'
  navigation_replace: true
name: View A
```

## Additional information

- This PR is related to issue or discussion: https://community.home-assistant.io/t/parametr-to-control-back-history-in-navigate-action/471028
- Link to documentation pull request: https://github.com/home-assistant/home-assistant.io/pull/28742

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [x] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by breakthestatic** [comment]
> Many users will not use this feature so we don't want to have it visible for everyone.

Makes sense to me! Would you prefer to revert the switch to ha-form & schema in hui-action-editor? Or does it make sense to leave it (with the `navigation_replace` removed).

Personally, I like how it handles most of the plumbing without option-specific code in the handlers, but obviously will defer to you guys!

**Comment by piitaya** [comment]
It's ok for the change ha-form 🙂 It's cleaner now!

File: src/panels/lovelace/components/hui-action-editor.ts [hunk 1]
```diff
@@ -203,10 +206,6 @@ export class HuiActionEditor extends LitElement {
         data = { service: this._service };
         break;
       }
-      case "navigate": {
-        data = { navigation_path: this._navigation_path };
-        break;
-      }
```
Comment (piitaya) [inline src/panels/lovelace/components/hui-action-editor.ts:209]: We should keep this for default value when selecting the navigate action.

File: src/panels/lovelace/components/hui-action-editor.ts [hunk 2]
```diff
@@ -65,11 +73,6 @@ export class HuiActionEditor extends LitElement {
 
   @property() protected hass?: HomeAssistant;
 
-  get _navigation_path(): string {
-    const config = this.config as NavigateActionConfig | undefined;
-    return config?.navigation_path || "";
-  }
-
```
Comment (piitaya) [inline src/panels/lovelace/components/hui-action-editor.ts:72]: We should keep this for default value when selecting the navigate action.

**Review by piitaya — CHANGES_REQUESTED** [review]
Thank you! I suggested a change to fix the editor, `navigation_path` shouldn't be a missing property so we need to keep the default value logic.

Can you also update the [documentation](https://www.home-assistant.io/dashboards/actions/#navigation_path) to add the new parameter.

**Comment by breakthestatic** [comment]
Thanks for the suggestions @piitaya! I restored the default value logic as requested. Also, you can find the docs update PR here - https://github.com/home-assistant/home-assistant.io/pull/28742. I wasn't sure if there was any convention for noting yaml-only vs gui options. If so, let me know and I will make the appropriate changes there as well.

**Review by piitaya — APPROVED** [review]
Thank you 🙂

**Pull request merged**